tests/v2/validation/provisioning/k3s/custom_cluster_test.go

```diff
@@ -85,71 +76,43 @@ func (c *CustomClusterProvisioningTestSuite) TestProvisioningK3SCustomCluster()
-		{"2 nodes - etcd/cp roles per 1 node " + provisioning.StandardClientName.String(), c.standardUserClient, nodeRolesShared, c.psact},
-		{"3 nodes - 1 role per node " + provisioning.AdminClientName.String(), c.client, nodeRolesDedicated, c.psact},
-		{"3 nodes - 1 role per node " + provisioning.StandardClientName.String(), c.standardUserClient, nodeRolesDedicated, c.psact},
+		{"1 Node all roles " + provisioning.AdminClientName.String(), c.client, nodeRolesAll},
+		{"1 Node all roles " + provisioning.StandardClientName.String(), c.standardUserClient, nodeRolesAll},
+		{"2 nodes - etcd/cp roles per 1 node " + provisioning.AdminClientName.String(), c.client, nodeRolesShared},
+		{"2 nodes - etcd/cp roles per 1 node " + provisioning.StandardClientName.String(), c.standardUserClient, nodeRolesShared},
+		{"3 nodes - 1 role per node " + provisioning.AdminClientName.String(), c.client, nodeRolesDedicated},
+		{"3 nodes - 1 role per node " + provisioning.StandardClientName.String(), c.standardUserClient, nodeRolesDedicated},
 	}
-	var name string
 	for _, tt := range tests {
-		testSession := session.NewSession()
-		defer testSession.Cleanup()
-
-		client, err := tt.client.WithSession(testSession)
-		require.NoError(c.T(), err)
-
-		for _, nodeProviderName := range c.nodeProviders {
-			externalNodeProvider := provisioning.ExternalNodeProviderSetup(nodeProviderName)
-			providerName := " Node Provider: " + nodeProviderName
-			for _, kubeVersion := range c.kubernetesVersions {
-				name = tt.name + providerName + " Kubernetes version: " + kubeVersion
-				c.Run(name, func() {
-					TestProvisioningK3SCustomCluster(c.T(), client, externalNodeProvider, tt.nodeRoles, kubeVersion, c.hardened, tt.psact, c.advancedOptions)
-				})
-			}
-		}
+		c.clustersConfig.NodesAndRoles = tt.nodeRoles
+		permutations.RunTestPermutations(&c.Suite, tt.name, tt.client, c.clustersConfig, permutations.K3SCustomCluster, nil, nil)
 	}
 }
 
 func (c *CustomClusterProvisioningTestSuite) TestProvisioningK3SCustomClusterDynamicInput() {
-	clustersConfig := new(provisioning.Config)
-	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
-	nodesAndRoles := clustersConfig.NodesAndRoles
-
-	if len(nodesAndRoles) == 0 {
+	isWindows := false
+	for _, pool := range c.clustersConfig.NodesAndRoles {
+		if pool.Windows {
+			isWindows = true
+			break
+		}
+	}
+	require.False(c.T(), isWindows, "Windows nodes are not supported for k3s Clusters")
+	if len(c.clustersConfig.NodesAndRoles) == 0 {
 		c.T().Skip()
 	}
 
+	poolNames := permutations.GetTestName(c.clustersConfig.NodesAndRoles)
 	tests := []struct {
 		name   string
 		client *rancher.Client
-		psact  string
 	}{
-		{provisioning.AdminClientName.String(), c.client, c.psact},
-		{provisioning.StandardClientName.String(), c.standardUserClient, c.psact},
+		{poolNames + provisioning.AdminClientName.String() + " Dynamic ", c.client},
```

Comment:
Let's keep the dynamic subtest names the same as before and remove all this extra. Doing this will make the regex calling super difficult and break a lot of our current testing in Jenkins which I would would want to avoid.